arc.go

```diff
@@ -177,30 +177,32 @@ func (c *ARCCache) Add(key, value interface{}) {
 
 // replace is used to adaptively evict from either T1 or T2
 // based on the current learned value of P
-func (c *ARCCache) replace(b2ContainsKey bool) {
+func (c *ARCCache[K, V]) replace(b2ContainsKey bool) {
 	t1Len := c.t1.Len()
 	if t1Len > 0 && (t1Len > c.p || (t1Len == c.p && b2ContainsKey)) {
 		k, _, ok := c.t1.RemoveOldest()
 		if ok {
-			c.b1.Add(k, nil)
+			var empty V
```

Comment:
Can you do me a favor and just explain this change? I haven't worked with generics a lot, but since `V` is of type `any`, isn't it perfectly valid to continue using `nil` here?

This will still compare against `nil` (tested it for my own benefit, https://go.dev/play/p/PH5XDYNs-Qu). So I don't think it's wrong, but curious about it.

Reply:
We can't keep `return nil, false`, because `nil` can't be assigned to all possible types of `V` (privimite types, non-pointer structs, etc.). This is also the reason for the other changes to naked returns.

https://go.dev/play/p/tf-1bsx02Q1

I can change the other return statements or add comments if you want.